tests/suites/test_suite_psa_crypto.function

```diff
@@ -6813,6 +7275,307 @@ exit:
+                PSA_ERROR_BAD_STATE );
+
+    PSA_ASSERT( psa_sign_hash_abort( &sign_operation ) );
+
+    PSA_ASSERT( psa_verify_hash_start( &verify_operation, key, alg,
+                                       input_data->x, input_data->len,
+                                       signature, signature_length ) );
+
+    TEST_EQUAL( psa_verify_hash_start( &verify_operation, key, alg,
+                                       input_data->x, input_data->len,
+                                       signature, signature_length ),
+                PSA_ERROR_BAD_STATE );
+
+    PSA_ASSERT( psa_verify_hash_abort( &verify_operation ) );
+
+    /* --- Ensure changing the max ops mid operation works (operation should
+     *     complete successfully after setting max ops to unlimited --- */
+    psa_interruptible_set_max_ops( 1 );
+
+    PSA_ASSERT( psa_sign_hash_start( &sign_operation, key, alg,
+                                     input_data->x, input_data->len ) );
+
+    TEST_EQUAL( psa_sign_hash_complete( &sign_operation, signature,
+                                        signature_size,
+                                        &signature_length ),
+                PSA_OPERATION_INCOMPLETE );
+
+    psa_interruptible_set_max_ops( PSA_INTERRUPTIBLE_MAX_OPS_UNLIMITED );
+
+    PSA_ASSERT( psa_sign_hash_complete( &sign_operation, signature,
+                                        signature_size,
+                                        &signature_length ) );
+
+    PSA_ASSERT( psa_sign_hash_abort( &sign_operation ) );
+
+    psa_interruptible_set_max_ops( 1 );
+
+    PSA_ASSERT( psa_verify_hash_start( &verify_operation, key, alg,
+                                       input_data->x, input_data->len,
+                                       signature, signature_length ) );
+
+    TEST_EQUAL( psa_verify_hash_complete( &verify_operation ),
+                PSA_OPERATION_INCOMPLETE );
+
+    psa_interruptible_set_max_ops( PSA_INTERRUPTIBLE_MAX_OPS_UNLIMITED );
+
+    PSA_ASSERT( psa_verify_hash_complete( &verify_operation ) );
+
+    PSA_ASSERT( psa_verify_hash_abort( &verify_operation ) );
+
+    /* --- Change function inputs mid run, to cause an error (sign only,
+     *     verify passes all inputs to start. --- */
+
+    psa_interruptible_set_max_ops( 1 );
+
+    PSA_ASSERT( psa_sign_hash_start( &sign_operation, key, alg,
+                                     input_data->x, input_data->len ) );
+
+    TEST_EQUAL( psa_sign_hash_complete( &sign_operation, signature,
+                                        signature_size,
+                                        &signature_length ),
+                PSA_OPERATION_INCOMPLETE );
+
+    TEST_EQUAL( psa_sign_hash_complete( &sign_operation, signature,
+                                        0,
+                                        &signature_length ),
+                PSA_ERROR_BUFFER_TOO_SMALL );
+
+    PSA_ASSERT( psa_sign_hash_abort( &sign_operation ) );
+
```

Comment:
[non-blocker] Missing test, to do in a follow-up:

* Good-case test for sign where `complete` runs at least twice, and `signature_size` remains the same but the `signature` buffer is different. Check both that the signature in the actual output buffer is valid (use a deterministic algorithm or do verification), and that the buffer passed to the first `complete` call is not used by the second call (I think the best way is, free it just before calling `complete` and rely on ASan).

Reply:
Filed as https://github.com/Mbed-TLS/mbedtls/issues/7108